tardis/tests/tests_slow/conftest.py

```diff
@@ -57,25 +50,31 @@ def plot_object(request):
     return PlotUploader(request)
 
 
-@pytest.fixture(scope="session")
-def integration_tests_config(request):
-    return request.config.option.integration_tests_config
+def setup_names():
+    slow_tests_dir = os.path.dirname(os.path.realpath(__file__))
+    names = list()
+    for child in os.listdir(slow_tests_dir):
+        if os.path.isdir(os.path.join(slow_tests_dir, child)):
+            names.append(child)
+    return names
 
 
-@pytest.fixture(scope="session")
-def reference_datadir(integration_tests_config):
-    return os.path.expandvars(
-        os.path.expanduser(integration_tests_config["reference"])
-    )
+@pytest.fixture(scope="class", params=setup_names())
+def data_paths(request):
```

Comment:
I think `data_paths` is a confusing name, as the fixture will always hold only one config set.

If we are returning a dictionary, then we can add the name. This removes the necessity to extract the name from the path later.